app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/AudioRecordActivityTest.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.sdp.ratemyepfl.activity
+
+import android.Manifest
+import android.content.Intent
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.rule.GrantPermissionRule
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.utils.TestUtils
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.hamcrest.CoreMatchers.not
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@HiltAndroidTest
+class AudioRecordActivityTest {
+    lateinit var scenario: ActivityScenario<AudioRecordActivity>
+
+    @get:Rule(order = 0)
+    val hiltRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val testRule = ActivityScenarioRule(AudioRecordActivity::class.java)
+
+    @get:Rule(order = 2)
+    val permissionRule: GrantPermissionRule = GrantPermissionRule.grant(
+        Manifest.permission.RECORD_AUDIO
+    )
+
+    @Before
+    fun setUp() {
+        val intent =
+            Intent(ApplicationProvider.getApplicationContext(), AudioRecordActivity::class.java)

```

Comment:
You don't need to launch an intent and use a scenario if you use a rule I think.

Reply:
Oh yes you're right! I will remove it.